### PR TITLE
Networking of scripted Blendshapes

### PR DESCRIPTION
**Pull request by thoys**
This PR will enable the following scripting API functionality:
```
MyAvatar.hasScriptedBlendshapes {boolean} Blendshapes will be transmitted over the network if set to true.
MyAvatar.hasAudioEnabledFaceMovement {boolean} If set to true, voice audio will move the mouth Blendshapes while MyAvatar.hasScriptedBlendshapes is enabled.
```

### Prerequisites 
- This test requires two avatars (Interface clients) in the same domain.
- **Tester A** make sure to use an avatar with proper blendshapes. It works the best with the Artemis avatar which should be available from market.

### Test
1. **Tester A**: run the facialExpressions script from https://cdn.rawgit.com/thoys/hifi-content/346a2ae1e0a062619c7d25c48fe51adf58298ed9/Prototyping/FacialExpressions/facialExpressions.js
2. **Tester B**: watch User A's face closely and remember their Neutral expression.
3. **Tester A**: Set your view to mirror view to see your own face (shortcut `2` `View->Mirror` ). (For comparance, be aware that the view is mirrored)
4. **Tester A**: Change your facial expression to each one of the following emotions and confirm with **Tester B**:
'H' = smile
'J' = laugh
'K' = flirt
'L = sad
'V' = angry
'B' = fear
'M' = disgust
'N' = neutral
5. **Tester B**: Make sure that you see the emotion change every time. 
6. **Tester A**: Stop the script.
7. **Tester A**: Look at yourself, your face should have returned to neutral.
8. **Tester B**: Confirm that **Tester A** their face returned to their neutral expression

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/7807/app_12584_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/7807/HighFidelity-Beta-PR12584-90b77030d5c7da0e88454b503de783bb7e85a080.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/7807/HighFidelity-Beta-PR12584-90b77030d5c7da0e88454b503de783bb7e85a080.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/9095/app_12584_PR_b0187ecfd3dfdceb084884a72cc1a432f6018ebf.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9095/HighFidelity-Beta-PR12584-b0187ecfd3dfdceb084884a72cc1a432f6018ebf.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9095/HighFidelity-Beta-Interface-PR12584-b0187ecfd3dfdceb084884a72cc1a432f6018ebf.dmg

File: interface/src/avatar/MyHead.cpp [hunk 1]
```diff
@@ -46,29 +46,33 @@ void MyHead::simulate(float deltaTime) {
...
+                const int FUNNEL_BLENDSHAPE = 40;
+                const int SMILE_LEFT_BLENDSHAPE = 28;
+                const int SMILE_RIGHT_BLENDSHAPE = 29;
+                _transientBlendshapeCoefficients[JAW_OPEN_BLENDSHAPE] += _audioJawOpen;
+                _transientBlendshapeCoefficients[SMILE_LEFT_BLENDSHAPE] += _mouth4;
+                _transientBlendshapeCoefficients[SMILE_RIGHT_BLENDSHAPE] += _mouth4;
+                _transientBlendshapeCoefficients[MMMM_BLENDSHAPE] += _mouth2;
+                _transientBlendshapeCoefficients[FUNNEL_BLENDSHAPE] += _mouth3;
             }
+             applyEyelidOffset(getFinalOrientationInWorldFrame());
```
Comment (hyperlogic) [inline interface/src/avatar/MyHead.cpp]: indentation level is off here.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/9157/app_12584_PR_192e48a376026b565ce89b39992b6060fc48d1c6.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9157/HighFidelity-Beta-PR12584-192e48a376026b565ce89b39992b6060fc48d1c6.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9157/HighFidelity-Beta-Interface-PR12584-192e48a376026b565ce89b39992b6060fc48d1c6.dmg

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/9163/app_12584_PR_484e20ea55dc6027b06447e6ca56baeabed88f52.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9163/HighFidelity-Beta-PR12584-484e20ea55dc6027b06447e6ca56baeabed88f52.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9163/HighFidelity-Beta-Interface-PR12584-484e20ea55dc6027b06447e6ca56baeabed88f52.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9163/HighFidelity-Beta-PR12584-484e20ea55dc6027b06447e6ca56baeabed88f52.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9163/HighFidelity-Beta-Interface-PR12584-484e20ea55dc6027b06447e6ca56baeabed88f52.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/9201/app_12584_PR_9fe062251cae7f93f427cb860b41f0ff9b227899.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9201/HighFidelity-Beta-PR12584-9fe062251cae7f93f427cb860b41f0ff9b227899.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9201/HighFidelity-Beta-Interface-PR12584-9fe062251cae7f93f427cb860b41f0ff9b227899.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9201/HighFidelity-Beta-PR12584-9fe062251cae7f93f427cb860b41f0ff9b227899.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9203/HighFidelity-Beta-PR12584-bbd31d590bb5f465fa5f906ac7dbf756abf7af53.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9203/HighFidelity-Beta-Interface-PR12584-bbd31d590bb5f465fa5f906ac7dbf756abf7af53.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9203/HighFidelity-Beta-PR12584-bbd31d590bb5f465fa5f906ac7dbf756abf7af53.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9203/HighFidelity-Beta-Interface-PR12584-bbd31d590bb5f465fa5f906ac7dbf756abf7af53.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/9815/HighFidelity-Beta-PR12584-8b4298514e58530d9c5236cef450ad03431a7d0a.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9815/HighFidelity-Beta-PR12584-8b4298514e58530d9c5236cef450ad03431a7d0a.dmg

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9815/HighFidelity-Beta-Interface-PR12584-8b4298514e58530d9c5236cef450ad03431a7d0a.exe

**Comment by qligmaloney** [comment]
Currently testing

**Comment by qligmaloney** [comment]
FAIL @conklin94122 
While talking with an expression selected, pieces of texture raise out of avatar's chest (as shown in gif).  After killing script, user's mouth was stuck in an open position, audio could be heard by other users but mouth didnt move. Broken textures still raised out of chest after script was killed.
Log attached
hifi-snap-by-qligmaloney-on-2018-05-16_15-52-33
pasted image at 2018_05_16 03_55 pm
[hifi-log.txt](https://github.com/highfidelity/hifi/files/2011008/hifi-log.txt)

**Comment by thoys** [comment]
@qligmaloney 

Thanks, the broken weight-paint is a known problem with that avatar.

Will look into the stuck blendshape issue.

**Comment by hyperlogic** [comment]
This PR is superseeded by #13338

**Pull request closed without merging**